File: September 2024/September 18/min_dist.py
def minimum_distance(a):
    answer = len(a)
    temp = 0

    for x in range(len(a)):
        for y in range(len(a)):
            if (a[x] == a[y]) and (x!=y) and (y >x):
                temp = y-x
                print(temp)
                if temp < answer:
                    answer = temp
                    temp = 0
    
    if answer == len(a):
        return -1
    else:
        return answer
    
print(minimum_distance([7, 1, 3, 4, 1, 7]))

File: September 2024/September 18/minDist.php
<?php

function minimumDistance($a){
    $answer = count($a);
    $temp = 0;

    for($i = 0; $i < count($a); $i++){
        for($j = $i+1; $j <count($a); $j++){
            
            if($a[$i]===$a[$j]){
                $temp = $j-$i;
                if($temp < $answer){
                    $answer = $temp;
                    $temp = 0;
                }
            }
        }
    }

    if($answer === count($a)){
        return -1;
    }else{
        return $answer;
    }

}

echo minimumDistance([7, 1, 3, 4, 1, 7]);